guidelines/configuration-api-design.md

```diff
@@ -0,0 +1,36 @@
+## Designing configuration API's
+
+All configuration API's should be code first since that allows us to best guide the users and allows us to evolve the API's using obsolete messages.
+
+Configuration API's should be variable based and not returning `this`.
+
+Example:
+
+```
+var myFeatureConfig = endpointConfig.EnableMyFeature()
+
+myFeatureConfig.SomeOption(X);
+myFeatureConfig.SomeOtherOption(X);
+
+```
+
+We prefer this over lambda based API's since:
```

Comment:
I hope we DO NOT rework the recoverability APIs but instead stick to the design guidelines for new APIs or when we have reasons to change this. The value of aligning it is not worth breaking the user code again